master/usage/assigning-ip-addresses-topology.md

```diff
@@ -0,0 +1,145 @@
+---
+title: Assigning IP addresses based on topology
+---
+
+## Why this is useful
+
+{{site.prodname}} can be configured to use specific IP pools for different topological areas.
+For example, you may want workloads in a particular rack, zone, or region to receive addresses from the same IP pool.
+This may be desirable either to reduce the number of routes required in the network or to meet requirements imposed by an external firewall device or policy.
+This guide walks through an example of how to set this up.
+
+## How this works
+At a high level, this feature is operated by the setting node labels and then selecting those
+nodes via node selector on your IP pool resources. Note that there are a variety of other ways
+to specify IP address assignment behavior that can be found in the
+[cni-plugin configuration reference document]({{site.baseurl}}/{{page.version}}/reference/cni-plugin/configuration).
+Each approach has its own use-case but this particular one takes on the lowest priority
+compared to the others (i.e. Specifying IP pools through configuration has precendence
+over IP addresses through annotations which has precendence over this IP pool
+assignment through node selectors).
+
+## Prerequisites
+
+**{{site.prodname}} IPAM**
+
+This guide only applies if you are using {{site.prodname}} IPAM.
+
+### Example: Kubernetes
+
+In this example, we created a cluster with four nodes across two racks (two nodes/rack). Consider the following:
+
+```
+       -------------------
+       |    router       |
+       -------------------
+       |                 |
+---------------   ---------------
+| rack-0      |   | rack-1      |
+---------------   ---------------
+| kube-node-0 |   | kube-node-2 |
+- - - - - - - -   - - - - - - - -
+| kube-node-1 |   | kube-node-3 |
+- - - - - - - -   - - - - - - - -
+```
+
+Using the pods IP range `192.168.0.0/16`, we target the following setup: reserve the `192.168.0.0/24` and `192.168.1.0/24`
+pools for `rack-0`, `rack-1`, `rack-2`, and `rack-3` respectively. Let's get started.
+
+
+By installing {{ site.prodname }} without setting the
+default IP pool to match, running `calicoctl get ippool -o wide` shows that {{site.prodname}} 
+created its default IP pool of `192.168.0.0/16`:
+
+```
+NAME                  CIDR             NAT    IPIPMODE   DISABLED
+default-ipv4-ippool   192.168.0.0/16   true   Always     false
+```
+{: .no-select-button}
+
+1. Delete the default IP pool.
+
+	Since the `default-ipv4-ippool` IP pool resource already exists and accounts for the entire `/16` block, we will have to delete this first:
+
+   ```
+   calicoctl delete ippools default-ipv4-ippool
+   ```
+
+2. Label the nodes.
```

Comment:
Do we need to mention about not doing this in prod and/or restarting all pods after doing this?

Reply:
We have some language about not doing things like this (I assume this is one of those changes that causes all the nodes to restart) on a "production cluster" or else, in a "maintenance window". Let me know if you would like some verbiage around this.